server/src/main/java/org/apache/iotdb/db/qp/physical/crud/BatchInsertPlan.java

```diff
@@ -288,6 +292,23 @@ public long getMaxTime() {
     return maxTime;
   }
 
+  public void updateMNodeLastValues(MNode node) {
+    long maxTime = Long.MIN_VALUE;
+    int maxIndex = 0;
+    for (int i = 0; i < times.length; i++) {
+      if (times[i] > maxTime) {
+        maxTime = times[i];
+        maxIndex = i;
+      }
+    }
+    for (int i = 0; i < measurements.length; i++) {
+      if (node.hasChild(measurements[i])) {
+        Object[] column = (Object[]) columns[i];
+        node.getChild(measurements[i]).updateCachedLast(maxTime, column[maxIndex], dataTypes[i]);
+      }
+    }
+  }
+
```

Comment:
This function is not the InsertPlan's responsibility

Reply:
Agree. Moved it out of InsertPlan